sycl/source/detail/queue_impl.hpp

```diff
@@ -96,11 +96,8 @@ class queue_impl {
         MDiscardEvents(
             has_property<ext::oneapi::property::queue::discard_events>()),
         MIsProfilingEnabled(has_property<property::queue::enable_profiling>()),
-        MHasDiscardEventsSupport(
-            MDiscardEvents &&
-            (MHostQueue ? true
-                        : (MIsInorder && getPlugin().getBackend() !=
-                                             backend::ext_oneapi_level_zero))) {
+        MHasDiscardEventsSupport(MDiscardEvents &&
```

Comment:
Why is discard-event only supported fir in-order?

Reply:
I think we can enable support of discard-event for out-of-order queues as well, I can't think of a reason against this idea.
But it is not an intention of this PR so I will prefer to do this separately because it may expose unexpected issues or cause some long discussion which will delay merging of this PR which is currently blocking another PR.

Reply:
OK